### PR TITLE
Bump io-sim bounds

### PR DESCRIPTION
**Pull request by dnadales**
# Checklist

- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Commits have useful messages
    - [ ] The documentation has been properly updated
    - [ ] New tests are added if needed and existing tests are updated
    - [ ] Any changes affecting Consensus packages must have an entry in the appropriate `changelog.d` directory created using [`scriv`](https://github.com/input-output-hk/scriv). If in doubt, see the [Consensus release process](../ouroboros-consensus/docs/ReleaseProcess.md).
    - [ ] Any changes in the Consensus API (every exposed function, type or module) that has changed its name, has been deleted, has been moved, or altered in some other significant way must leave behind a `DEPRECATED` warning that notifies downstream consumers. If deprecating a whole module, remember to add it to `./scripts/ci/check-stylish.sh` as otherwise `stylish-haskell` would un-deprecate it.
    - [ ] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [ ] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [ ] Reviewer requested

### CODE REVIEW

File: ouroboros-consensus-diffusion/ouroboros-consensus-diffusion.cabal [hunk 1]
```diff
@@ -88,7 +88,7 @@ library
     , filepath
     , fs-api
     , hashable
-    , io-classes                   ^>=0.3
+    , io-classes                   ^>=0.3     || ^>=0.5
```
Comment (coot) [inline ouroboros-consensus-diffusion/ouroboros-consensus-diffusion.cabal:91]:
```suggestion
    , io-classes                   ^>=0.3 && <0.6
```

Comment (coot) [reply ouroboros-consensus-diffusion/ouroboros-consensus-diffusion.cabal:91]: :thinking: or maybe you want to be more conservative in the consensus side?

Comment (dnadales) [reply ouroboros-consensus-diffusion/ouroboros-consensus-diffusion.cabal:91]: I think so :)

**Comment by coot** [comment]
`io-sim-0.6` is in `CHaP`.  Early next week I'd like to release `io-sim-1.0` then this PR will be obsoleted by #4281.

**Comment by dnadales** [comment]
Superseded by https://github.com/input-output-hk/ouroboros-network/pull/4281

**Pull request closed without merging**